SpeedrunTimer/data/tables/speedrun-timer-sct.tbm

```diff
@@ -0,0 +1,149 @@
+#Conditional Hooks
+$Application: FS2_Open
+
+$On Game Init:
+[
+--[[
+***Speedrun Timer - LiveSplit Integration***
+        By Zero Serenity
+
+Allows for timer integrations with LiveSplit for Freespace 1 and Freespace 2.
+
+]]--
+
+    SpeedrunTimer = {}
+
+    function SpeedrunTimer:Init()
+        self.Initialized = false
+        self.Started = false
+        self.RunningStatus = false --Status of the clock running.
+        self.SegmentTimes = {} --Holds segment times for the timer.
+        self.CurrentSegmentTime = 0 --The time for the current segment.
+        self.TotalTime = 0 --The time since the start of the run.
+        self.CurrentMission = "" --Holds the current mission filename.
+        self.LSEndpoint = "\\\\.\\pipe\\LiveSplit" --Localhost LiveSplit pipe. Will not likely work with a file.
+        self.InputOutput = require "io" --io is defaulted to the controls. Since we need it to send LiveSplit commands, we need an alias.
+        self.LSPipe = self.InputOutput.open(self.LSEndpoint, "w") --Open/start the pipe. Flush is required after every command.
+        ba.println("Speedrun Timer: Initialized - " .. tostring(time.getCurrentTime()))
+        self.Initialized = true
+    end
+
+    function SpeedrunTimer:StopTimer()
+        self.RunningStatus = false
+        ba.println("Speedrun Timer: StopTimer called - Last Segment Time: " .. self.CurrentSegmentTime)
+        self.CurrentSegmentTime = 0
+        self:PauseLiveSplit();
+    end
+
+    function SpeedrunTimer:PauseLiveSplit() --Send Pause to LiveSplit, and split if its on a tracked segment.
+        ba.println("Speedrun Timer: PauseLiveSplit called - Mission Filename: " .. mn.getMissionFilename())
+        if not(mn.isTraining() or --Training Missions
+         mn.getMissionFilename():upper() == "SM1-01" or --Start of the game.
+         (self.CurrentMission == mn.getMissionFilename() and --We restarted a mission...
+         not (ca.getNextMissionFilename() == nil)) --...and is not the end of the campaign.
+        ) then
+            self.LSPipe:write "split\n" --Split time.
+            self.LSPipe:flush()
+        end
+        self.LSPipe:write "pausegametime\n" --Stop the clock for the loading screen.
+        self.LSPipe:flush()
+        self.CurrentMission = mn.getMissionFilename()
+    end
+
+    function SpeedrunTimer:StartTimer()
+        if(ca == nil) then
```

Comment:
"ca" is a library, so I don't think it is ever nil.  There is a `mn.isInCampaign()` function if that's what you want.